docs/user/implement_env.rst

```diff
@@ -1,50 +1,68 @@
 .. _implement_mdp:
 
-===========================
-Adding a New Environment
-===========================
+================================
+Working with Garage Environments
+================================
 
-Garage uses an environment API based on the very popular OpenAI Gym interface. The main difference is that garage uses
-:code:`akro` to describe input and output spaces, which are an extension of the :code:`gym.Space` API.
+Garage uses the :code:`Environment` API for all RL environments.
 
-If you have an OpenAI Gym compatible environment, you can wrap it in :code:`garage.envs.GarageEnv` to use it with
-garage.
+The public API methods of :code:`Environment` are:
 
-.. code-block:: python
+* :code:`reset()`: Resets the environment. Return the first observation, and the episode-level information.
+* :code:`step()`: Steps the environment with an action and returns an EnvStep.
+
+    An :code:`EnvStep` is a named tuple of :code:`env_spec`, :code:`observation`, :code:`action`, :code:`reward`,
+    :code:`step_type` representing a single step in the environment.
+* :code:`render()`: Returns value for rendering, could be a terminal text, an image, etc.
+* :code:`visualize()`: Creates a visualization of the environment.
+* :code:`close()`: Closes the environment.
+
+The attributes of :code:`Environment` are:
 
-    import gym
-    from garage.envs import GarageEnv
+* :code:`action_space`: The action space specification.
+* :code:`observation_space`: The observation space specification.
+* :code:`spec`: The environment specifications.
+* :code:`render_modes`: The list of supported render modes.
 
-    myEnv = GarageEnv(gym.make('CartPole-v1'))  # Wraps a CartPole-v1 environment
+You should expect to use all environments in garage with this API.
 
 +++++++++++++++++++++++++++
-Add a Environment Wrapper
+Add an Environment Wrapper
```

Comment:
Usually environment wrapper means something different -- it's a class you wrap around environment to change something like the observation, action, etc.

This section should be more like "Add Support for an Existing Environment Library"